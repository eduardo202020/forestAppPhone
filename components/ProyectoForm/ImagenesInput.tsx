import React from 'react';
import {
  View,
  Pressable,
  Text,
  ScrollView,
  Image,
  StyleSheet,
  Modal,
} from 'react-native';

interface ImagenesInputProps {
  imagenes: string[];
  setModalVisible: (v: boolean) => void;
  modalVisible: boolean;
  horizontal?: boolean;
}

const ImagenesInput: React.FC<ImagenesInputProps> = ({
  imagenes,
  setModalVisible,
  modalVisible,
  horizontal = false,
}) => (
  <>
    <Text style={styles.label}>Imágenes</Text>
    {horizontal ? (
      <ScrollView horizontal style={styles.imagesRow}>
        {imagenes.map((img, idx) => (
          <Pressable
            key={idx}
            style={styles.imageBox}
            onPress={() => setModalVisible(true)}
          >
            <Image
              source={{ uri: img }}
              style={{ width: '100%', height: '100%', borderRadius: 10 }}
            />
          </Pressable>
        ))}
      </ScrollView>
    ) : (
      <View style={styles.imagesRow}>
        {imagenes.map((img, idx) => (
          <Pressable
            key={idx}
            style={styles.imageBox}
            onPress={() => setModalVisible(true)}
          >
            <Text style={{ color: '#888', textAlign: 'center' }}>
              Imagen {idx + 1}
            </Text>
          </Pressable>
        ))}
      </View>
    )}
    <Modal
      visible={modalVisible}
      transparent
      animationType="fade"
      onRequestClose={() => setModalVisible(false)}
    >
      <View style={styles.modalContainer}>
        <View style={styles.modalContent}>
          <Text>Funcionalidad para añadir imágenes próximamente</Text>
          <Text style={{ marginTop: 10, color: '#888' }}>
            (Próximamente podrás añadir o editar imágenes)
          </Text>
        </View>
      </View>
    </Modal>
  </>
);

const styles = StyleSheet.create({
  label: {
    fontWeight: 'bold',
    marginBottom: 4,
    marginTop: 8,
  },
  imagesRow: {
    flexDirection: 'row',
    gap: 8,
    marginBottom: 16,
  },
  imageBox: {
    width: 100,
    height: 100,
    backgroundColor: '#e0e0e0',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 8,
    marginBottom: 16,
  },
  modalContainer: {
    flex: 1,
    backgroundColor: 'rgba(0,0,0,0.3)',
    alignItems: 'center',
    justifyContent: 'center',
  },
  modalContent: {
    backgroundColor: '#fff',
    padding: 24,
    borderRadius: 10,
    alignItems: 'center',
  },
});

export default ImagenesInput;
