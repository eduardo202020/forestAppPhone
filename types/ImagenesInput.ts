export interface ImagenesInputProps {
  imagenes: string[];
  setModalVisible: (value: boolean) => void;
  modalVisible: boolean;
  horizontal?: boolean;
}
