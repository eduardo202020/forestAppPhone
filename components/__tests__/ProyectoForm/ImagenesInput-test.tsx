import React from 'react';
import renderer from 'react-test-renderer';
import ImagenesInput from '../../ProyectoForm/ImagenesInput';

describe('ImagenesInput', () => {
  it('se renderiza en modo horizontal', () => {
    const imagenes = ['https://test/1.png', 'https://test/2.png'];
    const setModalVisible = () => {};
    const tree = renderer
      .create(
        <ImagenesInput
          imagenes={imagenes}
          setModalVisible={setModalVisible}
          modalVisible={false}
          horizontal
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('se renderiza en modo vertical', () => {
    const imagenes = ['a', 'b'];
    const setModalVisible = () => {};
    const tree = renderer
      .create(
        <ImagenesInput
          imagenes={imagenes}
          setModalVisible={setModalVisible}
          modalVisible={false}
          horizontal={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
