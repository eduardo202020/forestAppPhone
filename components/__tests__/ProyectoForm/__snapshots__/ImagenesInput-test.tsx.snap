// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagenesInput se renderiza en modo horizontal 1`] = `
[
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginBottom": 4,
        "marginTop": 8,
      }
    }
  >
    Imágenes
  </Text>,
  <RCTScrollView
    horizontal={true}
    style={
      {
        "flexDirection": "row",
        "gap": 8,
        "marginBottom": 16,
      }
    }
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#e0e0e0",
            "borderRadius": 10,
            "height": 100,
            "justifyContent": "center",
            "marginBottom": 16,
            "marginRight": 8,
            "width": 100,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://test/1.png",
            }
          }
          style={
            {
              "borderRadius": 10,
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#e0e0e0",
            "borderRadius": 10,
            "height": 100,
            "justifyContent": "center",
            "marginBottom": 16,
            "marginRight": 8,
            "width": 100,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://test/2.png",
            }
          }
          style={
            {
              "borderRadius": 10,
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>,
]
`;

exports[`ImagenesInput se renderiza en modo vertical 1`] = `
[
  <Text
    style={
      {
        "fontWeight": "bold",
        "marginBottom": 4,
        "marginTop": 8,
      }
    }
  >
    Imágenes
  </Text>,
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 8,
        "marginBottom": 16,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e0e0e0",
          "borderRadius": 10,
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 16,
          "marginRight": 8,
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "#888",
            "textAlign": "center",
          }
        }
      >
        Imagen 
        1
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e0e0e0",
          "borderRadius": 10,
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 16,
          "marginRight": 8,
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "#888",
            "textAlign": "center",
          }
        }
      >
        Imagen 
        2
      </Text>
    </View>
  </View>,
]
`;
